src/test/java/com/contentgrid/gateway/runtime/routing/SimpleContentGridRequestRouterTest.java

```diff
@@ -0,0 +1,76 @@
+package com.contentgrid.gateway.runtime.routing;
+
+import static com.contentgrid.gateway.test.assertj.MonoAssert.assertThat;
+
+import com.contentgrid.gateway.runtime.application.ApplicationId;
+import com.contentgrid.gateway.runtime.application.ContentGridApplicationMetadata;
+import com.contentgrid.gateway.runtime.application.DeploymentId;
+import com.contentgrid.gateway.runtime.application.ServiceTracker;
+import com.contentgrid.gateway.runtime.application.SimpleContentGridApplicationMetadata;
+import com.contentgrid.gateway.runtime.application.SimpleContentGridDeploymentMetadata;
+import com.contentgrid.gateway.runtime.config.kubernetes.KubernetesLabels;
+import java.util.Map;
+import java.util.UUID;
+import lombok.NonNull;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.cloud.client.DefaultServiceInstance;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.mock.http.server.reactive.MockServerHttpRequest;
+import org.springframework.mock.web.server.MockServerWebExchange;
+
+class SimpleContentGridRequestRouterTest {
+
+    protected ContentGridApplicationMetadata applicationMetadata;
+    protected ServiceTracker serviceTracker;
+
+    protected ContentGridRequestRouter requestRouter;
+
+    @BeforeEach
+    void setup() {
+        var deployMetadata = new SimpleContentGridDeploymentMetadata();
+        this.applicationMetadata = new SimpleContentGridApplicationMetadata(deployMetadata);
+        this.serviceTracker = new ServiceTracker(event -> { }, applicationMetadata);
+        this.requestRouter = new SimpleContentGridRequestRouter(this.serviceTracker, this.applicationMetadata, deployMetadata);
+    }
+
+    @Test
+    void test() {
+        var deployId = DeploymentId.random();
+        var appId = ApplicationId.random();
+
+        var exchange1 = createExchange("https://{appId}.userapps.contentgrid.com/me", appId);
+        assertThat(this.requestRouter.route(exchange1)).isEmptyMono();
+
+
+        var service = serviceInstance(deployId, appId);
+
+        this.serviceTracker.handleServiceAdded(service);
+
+        var exchange2 = createExchange("https://{appId}.userapps.contentgrid.com/me", appId);
```

Comment:
Isn't this the same URL twice?

Is an exchange mutated by the requestRouter, or why do you create a new instance?

Reply:
Yes, it is the same url, but now there is a backing service available.
Yes, `ServerWebExchanges` get mutated passing through the `WebFilterChain` (but not by our requestRouter directly)